pulsarlib/messaging.go

```diff
@@ -143,8 +151,16 @@ func (m *messaging) processMessageWorker() {
 			Value:      messageItem.message.Payload(),
 			Properties: messageItem.message.Properties(),
 		}
-		messageItem.handler.HandleMessage(m)
-		messageItem.pulsarc.Ack(messageItem.message)
+		retry := messageItem.handler.HandleMessage(m)
+		if retry != nil {
+			if retry.RetryAfter == 0 {
+				messageItem.pulsarc.Nack(messageItem.message)
+			} else {
+				messageItem.pulsarc.ReconsumeLater(messageItem.message, retry.RetryAfter)
+			}
+		} else {
+			messageItem.pulsarc.Ack(messageItem.message) // Error of diffrent type, so preserve the default of ack
```

Comment:
Remove the comment. We don't know if it was an error or not. For us it is a success

Reply:
Done